pkg/config/store.go

```diff
@@ -0,0 +1,150 @@
+package config
+
+import (
+	"os"
+	"sort"
+	"strings"
+
+	toml "github.com/pelletier/go-toml"
+	"github.com/spf13/cast"
+)
+
+// StoreOpts are functional options for a Store.
+type StoreOpts struct {
+	EnvWhitelist map[string]string
+	EnvLookup    func(key string) (string, bool)
+}
+
+// Store is the backend for Config data. It aggregates multiple sources (env vars, TOML document)
+// and is able to get/set/unset key(s) in the TOML document.
+type Store struct {
+	tree         *toml.Tree
+	envWhitelist map[string]string
+	envLookup    func(key string) (string, bool)
+}
+
+// NewStore creates a Store according to a TOML tree and functional options.
+func NewStore(tree *toml.Tree, opts StoreOpts) *Store {
+	if opts.EnvWhitelist == nil {
+		opts.EnvWhitelist = map[string]string{
+			keyURL:      "DCOS_URL",
+			keyACSToken: "DCOS_ACS_TOKEN",
+			keyTLS:      "DCOS_SSL_VERIFY",
+			keyTimeout:  "DCOS_TIMEOUT",
+		}
+	}
+	if opts.EnvLookup == nil {
+		opts.EnvLookup = os.LookupEnv
+	}
+	return &Store{
+		tree:         tree,
+		envWhitelist: opts.EnvWhitelist,
+		envLookup:    opts.EnvLookup,
+	}
+}
+
+// Get a config value from the Store.
```

Comment:
I would make this more explicit: `Get returns a value from the Store using a key`.